plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java

```diff
@@ -128,12 +128,17 @@ protected void doDestroy() throws Exception {
 
   @Override
   public int count(final String operation, final Map<String, String> parameters) {
+    Iterable<Repository> proxyRepositories = getProxyRepositories();
     final List<Integer> remoteCounts = passQueryToRemoteRepos(nugetQuery(operation, parameters),
-        getProxyRepositories(),
+        proxyRepositories,
         new CountFetcher(fetcher));
-
-    final int hostedCount = count(operation, parameters, getHostedRepositories());
```

Comment:
Actually to be fair, the `StorageTx` javadoc does mention this for the `findAssets` case:

https://github.com/sonatype/nexus-oss/blob/master/components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTx.java#L116